ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py

```diff
@@ -2329,3 +2329,37 @@ def test_tensorflow_unravel_index(
         indices=indices[0],
         dims=dims[0],
     )
+
+# sequence_mask
+@handle_frontend_test(
+    fn_tree="tensorflow.sequence_mask",
+    dtype_and_lens=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("integer"),
+        num_arrays=1,
+        min_value=1,
+        max_value=5,
+    ),
+    max_len=st.integers(min_value=5, max_value=1000),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_sequence_mask(
+    *,
+    dtype_and_lens,
+    max_len,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+    backend_fw,
+):
+    input_dtype, lens = dtype_and_lens
+    helpers.test_frontend_function(
+        input_dtypes=['int32'],
```

Comment:
I think this is not correct, shouldn't it be `input_dtype`?

Reply:
The input_dtype comes out to be int64. This throws an error when I later attempt to cast it to maxlen's dtype saying it's not possible to cast from int64 to int32. This is not an issue at all as every value in the input array is less than maxlen, so if maxlen fits into int32, then so should the input array.

Reply:
That's a fault with your value generation then. `helpers.get_dtypes("integer")` generate both `int32` and `int64`, you are trying to convince the test function that you are sending a dtype different from what you are actually sending. Which is not right